lib/facter/resolvers/linux/hostname.rb

```diff
@@ -96,9 +96,9 @@ def hostname_and_no_domain?(hostname, domain)
 
           def read_domain
             file_content = Facter::Util::FileHelper.safe_read('/etc/resolv.conf')
-            if file_content =~ /^domain\s+(\S+)/
+            if file_content =~ /^domain\s+([^.]\S+)/
               Regexp.last_match(1)
-            elsif file_content =~ /^search\s+(\S+)/
+            elsif file_content =~ /^search\s+([^.]\S+)/
```

Comment:
AFAICT The following would have the affect of stripping all trailing dot's
```suggestion
            elsif file_content =~ /^search\s+(\S+[^.])/
```

Reply:
@b4ldr I didn't follow this comment. Are you suggesting we strip trailing dots?

Reply:
@joshcooper 

> Are you suggesting we strip trailing dots as was done in #200
yes

> in addition to stripping the leading dot?
Im not sure if we need to do that, i think that would be a change in behaviour?  

> Or that stripping the leading and trailing dot are equivalent when we're talking about root domains
also yes :)

Reply:
Do we want to strip all trailing dots ? Since empty labels are an error, I would be wary of making changes that make their usage valid in Facter's context.

Reply:
Sorry for the delay in responding.

Reply:
Hi @b4ldr this makes sense to me and I was also able to verify a domain with a leading dot is valid. I'll incorporate your suggestions now. Thanks for the response and suggestion!

Reply:
Hi @b4ldr thanks for your suggestion, I was also able to verify a domain with a leading dot is valid, like `.example.org` but with your suggested changes, it causes `Regexp.last_match(1)` to be something unexpected.

For example, if resolv.conf had `search foo.bar example.com` with `foo.bar` as the domain, then `Regexp.last_match(1)` should be `"foo.bar"` but with your suggestions, it returns `"foo.bar "`

Reply:
ahh yes of course this is getting more and more complicated, how about the following?
```ruby
elsif file_content =~ /^search\s+\.?(\S+[^.\s])/
```

Reply:
the other options is to just use `Regexp.last_match(1).strip` that might be a bit more readable